model/DeliveryFactory.php

```diff
@@ -39,39 +40,50 @@ class DeliveryFactory extends ConfigurableService
 
     /**
      * Creates a new simple delivery
-     * 
+     *
      * @param core_kernel_classes_Class $deliveryClass
      * @param core_kernel_classes_Resource $test
      * @param string $label
      * @return \common_report_Report
      */
     public function create(core_kernel_classes_Class $deliveryClass, core_kernel_classes_Resource $test, $label) {
-        
+
         \common_Logger::i('Creating '.$label.' with '.$test->getLabel().' under '.$deliveryClass->getLabel());
-        
+
         $storage = new TrackedStorage();
-        
+
         $testCompilerClass = \taoTests_models_classes_TestsService::singleton()->getCompilerClass($test);
         $compiler = new $testCompilerClass($test, $storage);
-        
+
         $report = $compiler->compile();
```

Comment:
you should probably do your check on the properties before compiling the test, it will avoid some wierd report like
```
Test publishing failed because "Test Category" is empty.
Items Compilation
Published Planets and moons
```